### PR TITLE
PlutusData and lossless Aeson decoding

### PR DESCRIPTION
**Pull request by klntsky**
Please check the JS part very carefully.

### CODE REVIEW

**Review by ngua — APPROVED** [review]
The JSON manipulation and related stuff looks good. I still think it's a good idea to include the type/kind signatures and explicit imports for consistency with the rest of the codebase

I'm going to approve this anyway so it doesn't get held up though

File: src/QueryM.purs [hunk 1]
```diff
@@ -364,6 +364,7 @@ utxoQueryDispatch
   -> String
   -> Effect (Either Json.JsonDecodeError (Effect Unit))
 utxoQueryDispatch ref str = do
+  -- TODO: replace it with the new implementation in `Aeson`.
```
[the remark this answers is not in the document]
Comment (adamczykm) [reply src/QueryM.purs:367]: :+1:

**Review by adamczykm — CHANGES_REQUESTED** [review]
- We should now use the new machinery to decode all JSONs. Am I right?
- The `parseJsonStringifyNumbers` can now be removed maybe the entire Helpers module if `fromJustEff` can be replaced by an existing function.
- I think we should expand the test suite. Tests from  test/Helpers.purs stress the js algorithm's ability to handle edge cases in string vs number recognition. They can be reused.

**Comment by ngua** [comment]
@adamczykm We have a separate issue to replace the existing stringified numbers JSON stuff (#151), so we can address that in a separate PR I think

**Comment by adamczykm** [comment]
What about the tests?

**Comment by ngua** [comment]
@adamczykm We can just re-use the tests from the `Helpers` module when #151 gets done. I don't think we should block this PR on doing that though

**Review by adamczykm — APPROVED** [review]
Do you intend to use the 'comments PR'
